pkg/splunk/controller/configmap.go

```diff
@@ -38,8 +39,24 @@ func ApplyConfigMap(client splcommon.ControllerClient, configMap *corev1.ConfigM
 	err := client.Get(context.TODO(), namespacedName, &current)
 	var dataUpdated bool
 	if err == nil {
-		if !reflect.DeepEqual(configMap.Data, current.Data) {
-			scopedLog.Info("Updating existing ConfigMap")
+		if !reflect.DeepEqual(configMap.Data, current.Data) || forceUpdate {
+			scopedLog.Info("Updating existing ConfigMap", "forceUpdate", forceUpdate, "ResourceVerison", current.GetResourceVersion())
+			if forceUpdate {
+				// Add or Increment the revision label to force a new ResourceVersion for the ConfigMap
+				labels := current.GetLabels()
+				if labels == nil {
+					newLabels := make(map[string]string)
+					newLabels["revision"] = "1"
+					current.SetLabels(newLabels)
+				} else if val, ok := labels["revision"]; ok {
+					revision, _ := strconv.Atoi(val)
+					labels["revision"] = strconv.Itoa(revision + 1)
+					current.SetLabels(labels)
+				} else {
+					labels["revision"] = "1"
+					current.SetLabels(labels)
+				}
+			}
```

Comment:
I may be wrong, but AFAIK, doing a `UpdateResource()` always gives us a new revision. In that case, we don't need the extra labels?

Reply:
I thought so too but thats not the case.  If you call `UpdateResource()` with the same info, it doesn't update the revision.
```
jryb:AppFramework$ k get configmap/splunk-demo-s1-standalone-app-list -o=yaml
apiVersion: v1
data:
  app-list-local.yaml: |-
    splunk:
      app_paths_install:
        default:
          - "/init-apps/admin-apps/demo_my_dashboards.spl"
kind: ConfigMap
metadata:
  creationTimestamp: "2021-09-08T22:02:47Z"
  name: splunk-demo-s1-standalone-app-list
  namespace: default
  ownerReferences:
  - apiVersion: enterprise.splunk.com/v2
    controller: true
    kind: Standalone
    name: demo-s1
    uid: fa455462-2046-46a1-aeb3-70db12fa03a7
  resourceVersion: "68382724"
  selfLink: /api/v1/namespaces/default/configmaps/splunk-demo-s1-standalone-app-list
  uid: 21261b76-6e96-4e19-aaa9-8adbab9a9b6b
jryb:AppFramework$ ##### UPDATE S3
```

```
{"level":"info","ts":1631139348.5761461,"logger":"splunk.enterprise.handleAppRepoChanges","msg":"received App listing","kind":"Standalone","name":"demo-s1","namespace":"default","for App sources":1}
{"level":"info","ts":1631139348.5761857,"logger":"splunk.enterprise.AddOrUpdateAppSrcDeploymentInfoList","msg":"App change detected.  Marking for an update.","Called with length: ":1,"appName":"demo_my_dashboards.spl"}
{"level":"info","ts":1631139348.5796883,"logger":"splunk.reconcile.ApplyConfigMap","msg":"Updating existing ConfigMap","name":"splunk-demo-s1-standalone-app-list","namespace":"default","forceUpdate":true,"ResourceVerison":"68382724"}
{"level":"info","ts":1631139348.584546,"logger":"splunk.reconcile.UpdateResource","msg":"Updated resource","name":"splunk-demo-s1-standalone-app-list","namespace":"default"}
```

```
jryb:AppFramework$ k get configmap/splunk-demo-s1-standalone-app-list -o=yaml
apiVersion: v1
data:
  app-list-local.yaml: |-
    splunk:
      app_paths_install:
        default:
          - "/init-apps/admin-apps/demo_my_dashboards.spl"
kind: ConfigMap
metadata:
  creationTimestamp: "2021-09-08T22:02:47Z"
  name: splunk-demo-s1-standalone-app-list
  namespace: default
  ownerReferences:
  - apiVersion: enterprise.splunk.com/v2
    controller: true
    kind: Standalone
    name: demo-s1
    uid: fa455462-2046-46a1-aeb3-70db12fa03a7
  resourceVersion: "68382724"
  selfLink: /api/v1/namespaces/default/configmaps/splunk-demo-s1-standalone-app-list
  uid: 21261b76-6e96-4e19-aaa9-8adbab9a9b6b